### PR TITLE
Improve docs for slice::from_raw_parts

### PR DESCRIPTION
**Pull request by RalfJung**
Explain why the pointer must be non-null and aligned.

Cc @hsivonen

### CODE REVIEW

**Comment by pietroalbini** [comment]
Picking someone from libs... r? @sfackler

**Comment by RalfJung** [comment]
Also see https://github.com/rust-lang/rust/pull/51134 where I previously changed these docs.

File: src/libcore/slice/mod.rs [hunk 1]
```diff
@@ -3873,9 +3873,11 @@ unsafe impl<'a, T> TrustedRandomAccess for ExactChunksMut<'a, T> {
 /// valid for `len` elements, nor whether the lifetime inferred is a suitable
 /// lifetime for the returned slice.
 ///
-/// `p` must be non-null and aligned, even for zero-length slices, as is
-/// required for all references. However, for zero-length slices, `p` can be
-/// a bogus non-dereferencable pointer such as [`NonNull::dangling()`].
+/// `data` must be non-null and aligned, even for zero-length slices. The
```
Comment (sfackler) [inline src/libcore/slice/mod.rs]: Is this *the* reason or *a* reason?

Comment (RalfJung) [reply src/libcore/slice/mod.rs]: I think "A reason" is more correct. Will fix.

**Pull request merged**